core/dbt/adapters/base/connections.py

```diff
@@ -412,3 +413,14 @@ def execute(
         :rtype: Tuple[AdapterResponse, agate.Table]
         """
         raise dbt.exceptions.NotImplementedError("`execute` is not implemented for this adapter!")
+
+    def dry_run(self, sql: str) -> AdapterResponse:
```

Comment:
since 'run' generally refers to a `dbt run` command (imagine if we develop a `--dry-run` flag for dbt at some point) should we just call this 'explain' or 'validate_sql'?

Reply:
`validate_sql` makes sense to me! That captures the narrower goal of this method for now